### PR TITLE
Feat/database table size check

### LINKED ISSUE #2: [Check] Introduce check to detect large DB tables

### PR DESCRIPTION
**Pull request by alexloetscher95**
resolves #2 (this time really)

### CODE REVIEW

**Comment by alexloetscher95** [comment]
Warning: 90 MB
Critical: 100 MB

Also added /docs/03-defaults.md

File: README.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ requirements, like availability of PHP extensions.
 - **DoctrineMigrations:** Checks whether all Doctrine Migrations have been migrated.
 - **HostingSize:** Checks how much Disk space is used by this hosting.
 - **DatabaseSize:** Checks how much space the Database uses on this hosting.
+- **DatabaseTableSize:** Checks how much space the Database Tables uses on this hosting.
```
Comment (aarongerig) [inline README.md]:
```suggestion
- **DatabaseTableSize:** Checks how much space each Database Table uses on this hosting.
```

File: src/PimcoreMonitorBundle/Check/DatabaseTableSize.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+     */
+    public function check(): ResultInterface
+    {
+        if ($this->skip) {
+            return new Skip('Check was skipped');
+        }
+
+        $sizes = $this->getDatabaseTableSizes();
+
+        if (!is_array($sizes)) {
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/Check/DatabaseTableSize.php]:
```suggestion
        if (!\is_array($sizes)) {
```

File: src/PimcoreMonitorBundle/Check/DatabaseTableSize.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            return new Failure('Database table sizes could not be retrieved');
+        }
+
+        $data = [
+            'ok' => 0,
+            'warning' => [],
+            'critical' => [],
+        ];
+
+        foreach($sizes as $size) {
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/Check/DatabaseTableSize.php]:
```suggestion
        foreach ($sizes as $size) {
```

File: src/PimcoreMonitorBundle/Check/DatabaseTableSize.php [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            }
+
+            ++$data['ok'];
+        }
+
+        if (\count($data['critical']) > 0) {
+            return new Failure(
+                \sprintf(
+                    'Following database table sizes are too high: %s',
+                    \implode(',', array_keys($data['critical']))),
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/Check/DatabaseTableSize.php]:
```suggestion
                    \implode(',', \array_keys($data['critical']))),
```

File: src/PimcoreMonitorBundle/Check/DatabaseTableSize.php [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                $data
+            );
+        }
+
+        if (\count($data['warning']) > 0) {
+            return new Warning(
+                \sprintf(
+                    'Following database table sizes are high: %s',
+                    \implode(',', array_keys($data['warning']))),
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/Check/DatabaseTableSize.php]:
```suggestion
                    \implode(',', \array_keys($data['warning']))),
```

**Review by aarongerig — CHANGES_REQUESTED** [review]

**Review by aarongerig — APPROVED** [review]

**Pull request merged**